src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -455,6 +467,38 @@ private void getCanvasResponse(HttpServletRequest request,
 		}else {
 			url=canvasURL+pathInfo;
 		}
+
+		TcSessionData tc = (TcSessionData) request.getSession().getAttribute(TC_SESSION_DATA);
+		M_log.debug("TC Session Data: " + tc);
+
+		//useful for debugging
+		//		Enumeration attrs = request.getSession().getAttributeNames();
+		//		while(attrs.hasMoreElements()){
+		//			M_log.debug("Attribute: " + attrs.nextElement());
+		//		}
+
+		//useful for debugging
+		//		Iterator tcIterator = tc.getCustomValuesMap().entrySet().iterator();
+		//		while(tcIterator.hasNext()){
+		//			Map.Entry pair = (Map.Entry)tcIterator.next();
+		//			M_log.debug(pair.getKey() + " = " + pair.getValue());
+		//		}
+
+		String stringToReplace = "user=self";
+
+		if (tc != null){
+			String uniqname = (String) tc.getCustomValuesMap().get("custom_canvas_user_login_id");
+			M_log.debug("uniqname: " + uniqname);
+
+			String replaceValue = "as_user_id=sis_login_id:" + uniqname;
+
+			if(url.toLowerCase().contains(stringToReplace.toLowerCase())){
+				url = url.replace(stringToReplace, replaceValue);
```

Comment:
If you need to use lowercase in the test, don't you need lower case in the replace?